### PR TITLE
WT-12376 Optimize history store size limiting code

### PR DESCRIPTION
**Pull request by bzm-jas**
Added a default value check for maximum history store size to ensure file size is only retrieved if it has been defined in the config.

### CODE REVIEW

File: src/history/hs_rec.c [hunk 1]
```diff
@@ -723,13 +723,16 @@ __wt_hs_insert_updates(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_MULTI *mult
...
     }
 
-    WT_ERR(__wt_block_manager_named_size(session, WT_HS_FILE, &hs_size));
     hs_btree = __wt_curhs_get_btree(hs_cursor);
     max_hs_size = hs_btree->file_max;
-    if (max_hs_size != 0 && (uint64_t)hs_size > max_hs_size)
-        WT_ERR_PANIC(session, WT_PANIC,
-          "WiredTigerHS: file size of %" PRIu64 " exceeds maximum size %" PRIu64, (uint64_t)hs_size,
-          max_hs_size);
+    /* Retrieve file size if history store size is enforced. */
```
Comment (etienneptl) [inline src/history/hs_rec.c:728]:
```suggestion
    /* Check the history store size if a limit is enforced. */
```

**Review by etienneptl — COMMENTED** [review]
The code changes look good to me!
@korteland, if this parameter does not seem to be used by MongoDB, do we want to keep it?

**Review by korteland — APPROVED** [review]
LGTM, and I agree with Etienne's suggestion for the comment.

**Comment by korteland** [comment]
> if this parameter does not seem to be used by MongoDB, do we want to keep it?

@etienneptl good call - I'll add it to the suggestions in the doc.

**Comment by bzm-jas** [comment]
The case where file size exceeds the enforced history store maximum value was tested using `hs_cleanup.cpp`. Self defined maximum value was set to 100MB (the minimum max file size) and the test successfully triggered WT_PANIC.

**Review by etienneptl — APPROVED** [review]
LGTM!

**Review by agorrod — APPROVED** [review]
LGTM - thanks!

**Pull request merged**